frontend/pages/invitation.tsx

```diff
@@ -0,0 +1,137 @@
+import { useEffect } from 'react';
+
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import Image from 'next/image';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+
+import { withLocalizedRequests } from 'hoc/locale';
+
+import { InferGetStaticPropsType } from 'next';
+
+import useMe from 'hooks/me';
+
+import { loadI18nMessages } from 'helpers/i18n';
+
+import Alert from 'components/alert';
+import Button from 'components/button';
+import Loading from 'components/loading';
+import { Paths, UserRoles } from 'enums';
+import { PageComponent } from 'types';
+
+import { useAcceptInvitation, useInvitedUser } from 'services/invitation/invitationService';
+
+export const getStaticProps = withLocalizedRequests(async ({ locale }) => {
+  return {
+    props: {
+      intlMessages: await loadI18nMessages({ locale }),
+    },
+  };
+});
+
+type InvitationProps = InferGetStaticPropsType<typeof getStaticProps>;
+
+const Invitation: PageComponent<InvitationProps> = () => {
+  const { formatMessage } = useIntl();
+  const { push, query, replace } = useRouter();
+  const {
+    invitedUser,
+    isError: invitedUserError,
+    isLoading: invitedUserLoading,
+  } = useInvitedUser(query.invitation_token as string);
+  const { user, isError: userError, isLoading: userLoading } = useMe();
+  const acceptInvitation = useAcceptInvitation();
+
+  useEffect(() => {
+    if (
+      (!!invitedUser && !!user && user.email !== invitedUser.email) ||
+      (!!user && user?.role !== UserRoles.Light)
+    ) {
+      replace(Paths.Dashboard);
+    } else if (!!invitedUser && userError) {
+      // If the user has an user account but is not signed in
+      replace({
+        pathname: invitedUser.requires_registration ? Paths.SignUp : Paths.SignIn,
+        query: { invitation_token: query.invitation_token },
+      });
+      return null;
+    }
+  }, [invitedUser, query.invitation_token, replace, user, userError]);
+
+  const handleAccept = () => {
+    acceptInvitation.mutate(query.invitation_token as string, {
+      onSuccess: () => push(Paths.Dashboard),
+    });
+  };
+
+  return !!user ? (
```

Comment:
I think you added this check because of the flash I was describing:
> If I'm a logged out user, when invitedUserLoading and userLoading are false, there's a very brief flash of the page before I get redirected to the sign up or sign in page.

Unfortunately, this check created one bug and an undesired behaviour:
- Log out and follow [this link](http://localhost:3000/invitation?invitation_token=TEi1FeWtPyNmYson-2Ts). You'll see you get absolutely no feedback about what's happening. If you remove the check however, you see the loading screen and then the error is displayed: “Invalid invitation token”.
- While the data is loading, there is nothing displayed to the user. I think it was better when we had the loading spinner.

Do you think we could remove this check? You can ignore my previous feedback about the flash. It's not critical.